### PR TITLE
Allow to set custom properties

### LINKED ISSUE #199: Be Able to Add Custom Properties (Key/Value) through Properties Panel
__Is your feature request related to a problem? Please describe__

Allow to add custom properties (key/value) via propertiesPanel.

__Describe the solution you'd like__

**What**
In the form-js editor, in the properties panel allow to add custom properties (key value). 
Persist these custom properties in the resulting `form.json`.

**Example for form.json**
```
{
  "schemaVersion": 2,
  "components": [
    {
      "label": "Text Field",
      "type": "textfield",
      "id": "Field_0ybooxt",
      "key": "field1",
      "customProperties": 
        {
           "key1": "value1",
           "key2": "value2"
        }
    },
    {
      "label": "Text Field",
      "type": "textfield",
      "id": "Field_09h5omo",
      "key": "field2"
    }
  ],
  "type": "default",
  "id": "Form_0aj08qi"
}
```

**Why**
These additionalProperties can be used by custom tasklist implementations to customize the form.
For example the customer has a use-case, where he/she populates a dropDown based on an external API. To know which custom API to call, the additionalProperties are used.

__Describe alternatives you've considered__

Workaround exists: add properties via JSON tab. Customer reported that: `this is not a suitable solution for business users, also prone to errors`

__Additional context__

See https://jira.camunda.com/browse/SUPPORT-11774

### PR DESCRIPTION
**Pull request by smbea**
Closes #199

### CODE REVIEW

**Review by MaxTru — CHANGES_REQUESTED** [review]
Quick smoke test: I get an error in the console when adding multiple values =>
Peek 2022-02-16 21-30

File: packages/form-js-editor/src/render/components/properties-panel/groups/CustomValuesGroup.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+      if (properties[value])
+        return 'Must be unique.';
+    };
+  };
+
+  const hasEntries = Object.keys(properties).length > 0;
+
+  return (
+    <Group label="Custom Properties" addEntry={ addEntry } hasEntries={ hasEntries }>
```
Comment (MaxTru) [inline packages/form-js-editor/src/render/components/properties-panel/groups/CustomValuesGroup.js:46]: Let's use sentence case in the label. So `Custom properties`.

**Comment by smbea** [comment]
> Quick smoke test: I get an error in the console when adding multiple values 

This is also the case for the radio and select values. I will try to figure out why this happens

**Comment by MaxTru** [comment]
> > Quick smoke test: I get an error in the console when adding multiple values
> 
> This is also the case for the radio and select values. I will try to figure out why this happens

Ah I see - thanks for the note. Then maybe seperate issue.

**Comment by smbea** [comment]
> Ah I see - thanks for the note. Then maybe seperate issue.

I gave it a quick look but couldn't fix it easily, so maybe I can create a  separate issue for this

**Comment by philippfromme** [comment]
> I gave it a quick look but couldn't fix it easily, so maybe I can create a separate issue for this

Let me have a look. Maybe it's an easy fix.

**Comment by smbea** [comment]
I believe this was fixed for the `custom properties` with my new push. I had a typo there which caused the error for `custom properties`.

But I didn't touch the `values` for select and radio - these are still causing the error. I think this isn't a problem for `custom properties` because of the key attribute being unique, unlike in `values`.

**Comment by philippfromme** [comment]
The issue is unrelated. It's about Preact keys. I'll look into it. It doesn't block this pull request.

**Review by philippfromme — APPROVED** [review]
I've added a couple of adjustments.

One thing we have to be careful about is mutating data that should not be mutated outside of command execution:

```javascript
const removeEntry = () => {
  delete properties[key]; // ⚠️⚠️⚠️
  editField(field, [ 'properties' ], properties);
};
```

cf. https://github.com/bpmn-io/form-js/pull/226/commits/ea06435016cd01a126b0f8366e393a13a93f43e7#diff-d9aa085d1d1aa4b6c29d41a645a753822e359cb36ac6d33ce0a7a6ffd138b3dbR51

cf. https://github.com/bpmn-io/form-js/pull/226/commits/ea06435016cd01a126b0f8366e393a13a93f43e7#diff-797efb599d143b3baeb4b8bdfd2c2c603de68ce1c64b55af7a6dbd4a561f610bR18

Furthermore, since we added a new property `properties` to the schema we need to bump the schema version as it's the convention for form-js (we don't use JSON schema).

cf. https://github.com/bpmn-io/form-js/pull/226/commits/560220c308e450b9bd42bfc7b667df7eae7ad0de

As for code style, we generally don't omit curly braces as it's error-prone:

```javascript
if (properties[value])
  return 'Must be unique.'; // ⚠️⚠️⚠️
```

cf. https://github.com/bpmn-io/form-js/pull/226/commits/ca6efe3f037817b8a4d128dff49cb69fa9254dac#diff-d9aa085d1d1aa4b6c29d41a645a753822e359cb36ac6d33ce0a7a6ffd138b3dbL39

Good job! 👏🏻

**Comment by philippfromme** [comment]
Maybe you can follow up with https://github.com/bpmn-io/form-js/issues/227.

**Comment by smbea** [comment]
@philippfromme thanks for the info! I'm working on something else at the moment, but I'll add that one to my radar for when I have time

**Review by MaxTru — APPROVED** [review]

**Pull request merged**